lib/Fmt_ast.ml

```diff
@@ -734,11 +734,12 @@ and fmt_core_type c ?(box = true) ?(in_type_declaration = false) ?pro
       | Some pro when c.conf.ocp_indent_compat ->
           fits_breaks ""
             (String.make (Int.max 1 (indent - String.length pro)) ' ')
-      | _ ->
+      | _ when box ->
           fmt_if_k
             Poly.(c.conf.break_separators = `Before)
             (fmt_or_k c.conf.ocp_indent_compat (fits_breaks "" "")
-               (fits_breaks "" "   ") ) )
+               (fits_breaks "" "   ") )
+      | _ -> noop )
```

Comment:
This fixes the problem only by chance I think. I get better result by passing `~box:true` in `Ptyp_poly` below, the initial indentation is not lost and I prefer how it breaks.

Reply:
Yes I've also noticed this, but I think the wrapping was a feature of the original formatting, that's why I preserved it.
Though I agree that breaking makes the output easier to read, I aimed at reducing the diff for the user.

Reply:
Didn't receive a reply from Josh so I guess whether or not it was intended we can change it, here is the test_branch diff for the janestreet profile for the record:
```diff
diff --git a/infer/src/absint/AnalysisCallbacks.ml b/infer/src/absint/AnalysisCallbacks.ml
index 95d6fb773..815966a69 100644
--- a/infer/src/absint/AnalysisCallbacks.ml
+++ b/infer/src/absint/AnalysisCallbacks.ml
@@ -11,8 +11,12 @@ module L = Logging
 type callbacks =
   { get_proc_desc_f : Procname.t -> Procdesc.t option
   ; html_debug_new_node_session_f :
-      'a. ?kind:[ `ComputePre | `ExecNode | `ExecNodeNarrowing | `WTO ]
-      -> pp_name:(Format.formatter -> unit) -> Procdesc.Node.t -> f:(unit -> 'a) -> 'a
+      'a.
+      ?kind:[ `ComputePre | `ExecNode | `ExecNodeNarrowing | `WTO ]
+      -> pp_name:(Format.formatter -> unit)
+      -> Procdesc.Node.t
+      -> f:(unit -> 'a)
+      -> 'a
   ; proc_resolve_attributes_f : Procname.t -> ProcAttributes.t option
   }
 
diff --git a/infer/src/absint/AnalysisCallbacks.mli b/infer/src/absint/AnalysisCallbacks.mli
index ad9d31a35..301eb4723 100644
--- a/infer/src/absint/AnalysisCallbacks.mli
+++ b/infer/src/absint/AnalysisCallbacks.mli
@@ -31,8 +31,12 @@ val proc_resolve_attributes : Procname.t -> ProcAttributes.t option
 type callbacks =
   { get_proc_desc_f : Procname.t -> Procdesc.t option
   ; html_debug_new_node_session_f :
-      'a. ?kind:[ `ComputePre | `ExecNode | `ExecNodeNarrowing | `WTO ]
-      -> pp_name:(Format.formatter -> unit) -> Procdesc.Node.t -> f:(unit -> 'a) -> 'a
+      'a.
+      ?kind:[ `ComputePre | `ExecNode | `ExecNodeNarrowing | `WTO ]
+      -> pp_name:(Format.formatter -> unit)
+      -> Procdesc.Node.t
+      -> f:(unit -> 'a)
+      -> 'a
   ; proc_resolve_attributes_f : Procname.t -> ProcAttributes.t option
   }
 
diff --git a/src/dune_engine/string_with_vars.ml b/src/dune_engine/string_with_vars.ml
index 70a828a55..b0b7e67d3 100644
--- a/src/dune_engine/string_with_vars.ml
+++ b/src/dune_engine/string_with_vars.ml
@@ -241,7 +241,11 @@ module Make (A : Applicative_intf.S1) = struct
   module App = Applicative.Make (A)
 
   let partial_expand :
-        'a. t -> mode:'a Mode.t -> dir:Path.t -> f:Value.t list option A.t expander
+        'a.
+        t
+        -> mode:'a Mode.t
+        -> dir:Path.t
+        -> f:Value.t list option A.t expander
         -> 'a Partial.t A.t
     =
    fun ({ template; syntax_version } as t) ~mode ~dir ~f ->
```